tardis/workflows/v_inner_solver.py

```diff
@@ -0,0 +1,353 @@
+            self.convergence_strategy.v_inner_boundary
+        )
+
+        if tau is not None:
+            self.TAU_TARGET = np.log(tau)
+
+        initial_v_inner = self.estimate_v_inner()
+
+        self.simulation_state.geometry.v_inner_boundary = initial_v_inner
+        self.simulation_state.blackbody_packet_source.radius = (
+            self.simulation_state.r_inner[0]
+        )
+
+    def estimate_v_inner(self):
+        """
+        Compute the Rosseland Mean Optical Depth,
+        Estimate location where v_inner makes t=2/3 (or target)
+        Extrapolate with exponential fits
+
+        Need some way to return and inspect the optical depths for later logging
+        """
+        tau_integ = np.log(
+            get_tau_integ(
+                self.plasma_solver,
+                self.simulation_state,
+            )[self.mean_optical_depth]
+        )
+
+        interpolator = interp1d(
+            tau_integ,
+            self.simulation_state.geometry.v_inner,  # Only use the active values as we only need a numerical estimate, not an index
+            fill_value="extrapolate",
+        )
+
+        estimated_v_inner = interpolator(self.TAU_TARGET) * u.cm / u.s
+
+        if estimated_v_inner < self.simulation_state.geometry.v_inner[0]:
+            estimated_v_inner = self.simulation_state.geometry.v_inner[0]
+            logger.warning(
+                "WARNING: v_inner_boundary outside of simulation, setting to first shell"
+            )
+        elif estimated_v_inner > self.simulation_state.geometry.v_inner[-1]:
+            estimated_v_inner = self.simulation_state.geometry.v_inner[-1]
+            logger.warning(
+                "WARNING: v_inner_boundary outside of simulation, setting to last shell"
+            )
+
+        return estimated_v_inner
+
+    @property
+    def property_mask(self):
+        mask = np.zeros(
+            (len(self.simulation_state.geometry.r_inner)), dtype=bool
+        )
+        mask[
+            self.simulation_state.geometry.v_inner_boundary_index : self.simulation_state.geometry.v_outer_boundary_index
+        ] = True
+        return mask
+
+    def get_convergence_estimates(self, transport_state):
+        """Compute convergence estimates from the transport state
+
+        Parameters
+        ----------
+        transport_state : MonteCarloTransportState
+            Transport state object to compute estimates
+
+        Returns
+        -------
+        dict
+            Convergence estimates
+        EstimatedRadiationFieldProperties
+            Dilute radiation file and j_blues dataclass
+        """
+        estimates = super().get_convergence_estimates(transport_state)
+
+        estimated_v_inner = self.estimate_v_inner()
+
+        estimates[0].update(
+            {"v_inner_boundary": estimated_v_inner, "mask": self.property_mask}
+        )
+
+        return estimates
+
+    def reproject(self, a1, m1, a2, m2):
```

Comment:
Not sure if this is would be used elsewhere, but it seems like quite a generic function to live in this class. Maybe this should live in util.py? 

Also, a1, m1, a2, and m2 could be a little more explicit. Just named "array1" and "mask1".